### DISCUSSION TITLE
Longhorn on AKS with azure disks as storage backend.

### DISCUSSION THREAD

**Discussion by michaelgroening** [longhorn/longhorn#9520] [Q&A]
Hi!

for a customer I am evaluating a solution to work around the limit of attached volumes on Azure AKS nodes. 
The customer hosts a bunch of PgSQL DBs on AKS and maps the PVCs of the deployments to Azure Disk Volumes.
As the number of mounts in Azure is limited by the number of CPUs, the customer is provisioning nodes with way more CPU than is needed to fullfill the workload. 
We are now evaluating options to solve this problem.
One solution would be to map volumes to azure Files via NFS but we are skeptical about which influence this will have on DB Availability and consistency.
Another option would be to Mount Azure Disk Volumes into Longhorn and then use Longhorn storage classes in the Postgres instances.

My Questions are:
has any of you tried this or something similar before?
If yes how did it work out for you? 
Is there anything to consider when deploying longhorn on AKS?

I am really looking forward to hear your thoughts about this idea.

Thanks for your support in advance!

**Comment by shuo-wu** [comment]
Yes. Physical/Cloud vendor disk - Longhorn volumes - DB is one typical work stack or use case for Longhorn.

One concern is, if there is a pretty high requirement over the performance.

**Reply by michaelgroening** [reply]
Integrity will be valued more than transaction throughput. As we will probably start with the smallest loads first. Thanks for the answer!

**Comment by broomfn** [comment]
+1, it seems odd that this doesn't seem to be an option out-of-the-box, from what I understand, it looks like longhorn relies on the OS disks in the nodes? What concern me slightly is data lose if the nodes vanish (e.g. upgrades, scaling etc.), I'm guessing the backups are needed for this in case.  As Longhorn relies on OS disks, increasing the storage requires migrating the node pool to a new pool with large OS disks, of which there is a complicated process, where you cannot use the Azure Portal, az cli or Terraform, but need to use Longhorn UI to evict and drain pools, before you delete the old pool, it all worries me that it's far to easy for an engineer to forget, get it wrong and before you know it you've lost your data and have to go though a complicated DR recovery process. Please correct me if I'm wrong or I've misunderstood 🤷‍♂️

**Comment by eh8** [comment]
This is a late reply, but @michaelgroening have you considered using Azure Container Storage? It's a new persistent storage platform for Kubernetes built for AKS.

With it, you can mount 75 PVs per node regardless of VM SKU.

Full disclosure: I work on Azure Container Storage
